plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerMetadata.java

```diff
@@ -391,11 +422,85 @@ public synchronized Optional<ConnectorOutputMetadata> finishCreateTable(
         TableInfo info = tables.get(tableName);
         requireNonNull(info, "info is null");
 
-        tables.put(tableName, new TableInfo(info.columns(), info.properties(), info.comment()));
+        tables.put(tableName, getTableUsingStats(info, computedStatistics));
 
         return Optional.empty();
     }
 
+    private synchronized TableInfo getTableUsingStats(TableInfo info, Collection<ComputedStatistics> computedStatistics)
+    {
+        if (computedStatistics.isEmpty()) {
+            return info;
+        }
+        ImmutableMap.Builder<String, Object> minimumsBuilder = ImmutableMap.builder();
+        ImmutableMap.Builder<String, Object> maximumsBuilder = ImmutableMap.builder();
+        ImmutableMap.Builder<String, Long> distinctValuesBuilder = ImmutableMap.builder();
+        List<ColumnInfo> columns = info.columns();
+        Map<String, Type> types = columns.stream().collect(toMap(ColumnInfo::name, ColumnInfo::type));
+        AtomicLong rowCount = new AtomicLong(1);
+        computedStatistics.forEach(statistic -> {
+            rowCount.set((long) firstNonNull(readNativeValue(BIGINT, statistic.getTableStatistics().get(ROW_COUNT), 0), 0));
+            statistic.getColumnStatistics().forEach((metadata, block) -> {
+                checkState(block.getPositionCount() == 1, "Expected a single position in aggregation result block");
+                String columnName = metadata.getColumnName();
+                Type type = types.get(columnName);
+                switch (metadata.getStatisticType()) {
```

Comment:
Don't we need `NUMBER_OF_NON_NULL_VALUES` as well to get nulls probability for the column ?

Reply:
Done!